src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,195 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from enum import StrEnum
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, TypeAdapter
+
+log = logging.getLogger(__name__)
+
+
+# Classes for ResourceQuotaSpec
+class Operator(StrEnum):
+    """Pydantic class for the Operator in MetchExpression of ResourceQuotaSpec."""
+
+    In = "In"
+    NotIn = "NotIn"
+    Exists = "Exists"
+    DoesNotExist = "DoesNotExist"
+
+
+class ScopedResourceSelectorRequirement(BaseModel, extra="forbid"):
+    """Pydantic class for objects of matchExpressions of ResourceQuotaSpec.
```

Comment:
You can avoid saying "Pydantic class", we normally don't need to explain what parent class we are inheriting from. You can just leave "Representation of a scoped resource" or something similar.

This comment applies to other classes.

Reply:
127608f